src/rat/rat_basin.py

```diff
@@ -495,43 +497,47 @@ def rat_basin(config, rat_logger):
                     basin_flow_direction_file = basin_flow_dir_file,
+                    rout_input_path_prefix = rout_input_path_prefix,
                     clean=False,
-                    rout_input_path_prefix = rout_input_path_prefix
-                    ) as r:
-                    route = RoutingRunner(    
-                        project_dir = config['GLOBAL']['project_dir'], 
-                        result_dir = r.params['output_dir'], 
-                        inflow_dir = rout_inflow_dir, 
-                        model_path = config['ROUTING']['route_model'],
-                        param_path = r.route_param_path, 
-                        fdr_path = r.params['flow_direction_file'], 
-                        station_path_latlon = basin_station_latlon_file,
-                        station_xy = r.params['station']
-                    )
-                    route.create_station_file()
-                    route.run_routing()
-                    route.generate_inflow()
-                    basin_station_xy_path = route.station_path_xy
+                    inflow_dir = rout_inflow_dir,
+                    station_path_latlon = basin_station_latlon_file,
+                )
                 ROUTING_STATUS=1
             else:
                 rat_logger.info("VIC Run Failed. Skipping Step-8: Runnning Routing and generating Inflow")
         except:
             no_errors = no_errors+1
             rat_logger.exception("Error Executing Step-8: Runnning Routing and generating Inflow")
         else:
-            rat_logger.info("Finished Step-8: Runnning Routing and generating Inflow")
-            #------------- Routing Ends and Inflow pre-processed for Mass Balance --------------#
+            rat_logger.info("Finished Step-8: Runnning Routing model")
+            #------------- Routing Ends --------------#
 
     ######### Step-9
     if(9 in steps):
         try:
-            rat_logger.info("Starting Step-9: Preparation of parameter files for Surface Area Calculation")
+            rat_logger.info("Starting Step-9: Generating Inflow files")
+            #-------------  Inflow pre-processing for Mass Balance begins--------------#
+            ###### Converting routing model's outputs to .csv file
+            routing_output_dir = Path(config['GLOBAL']['data_dir']).joinpath(config['BASIN']['region_name'], 'basins', basin_name, 'ro','ou')
+            dst_dir = Path(config['GLOBAL']['data_dir']).joinpath(config['BASIN']['region_name'], 'basins', basin_name, 'ro', 'rout_inflow')
+            dst_dir.mkdir(parents=True, exist_ok=True)
+            generate_inflow(routing_output_dir, dst_dir)
+        except:
+            no_errors = no_errors+1
+            rat_logger.exception("Error Executing Step-9: Generating Inflow files")
+        else:
+            rat_logger.info("Finished Step-9: Generating Inflow files")
+            #------------- Inflow pre-processing for Mass Balance ends--------------#
```

Comment:
Merge this with previous Step 13 (currently step 14) so that outflow, area, evaporation and inflow gets calculated in a single step.